conans/model/build_info.py

```diff
@@ -225,6 +244,120 @@ def framework_paths(self):
         return self.frameworkdirs
 
 
+class DepCppInfo(object):
+
+    def __init__(self, cpp_info):
+        self._cpp_info = cpp_info
+        self._libs = None
+        self._system_libs = None
+        self._frameworks = None
+        self._defines = None
+        self._cxxflags = None
+        self._cflags = None
+        self._sharedlinkflags = None
+        self._exelinkflags = None
+
+        self._include_paths = None
+        self._lib_paths = None
+        self._bin_paths = None
+        self._build_paths = None
+        self._res_paths = None
+        self._src_paths = None
+        self._framework_paths = None
+        self._build_module_paths = None
+
+    def __getattr__(self, item):
+        try:
+            attr = self._cpp_info.__getattribute__(item)
+        except AttributeError:  # item is not defined, get config (CppInfo)
+            attr = self._cpp_info.__getattr__(item)
+        return attr
+
+    @staticmethod
+    def _merge_lists(seq1, seq2):
+        return [s for s in seq1 if s not in seq2] + seq2
+
+    def _aggregated_values(self, item):
+        if getattr(self, "_%s" % item) is None:
+            values = getattr(self._cpp_info, item)
+            for _, component in self._cpp_info.components.items():
+                values = self._merge_lists(values, getattr(component, item))
+            setattr(self, "_%s" % item, values)
+        return getattr(self, "_%s" % item)
+
+    def _aggregated_paths(self, item):
+        if getattr(self, "_%s_paths" % item) is None:
+            values = getattr(self._cpp_info, "%s_paths" % item)
+            for _, component in self._cpp_info.components.items():
+                values = self._merge_lists(values, getattr(component, "%s_paths" % item))
```

Comment:
.... because maybe here we can provide the `self.rootpath`, no need to store a value inside (read the other comment above).

There are different approaches to implement this suggestion:
 1. The `Component` class takes the pointer to the parent instead of the `rootpath`: same problem as the one above, but at least it would be updated always (not sure if this is needed for the editable packages with layouts).
 2. Instead of a property, add functions that take the argument: `get_xxxx_paths(rootpath)`
 3. Do `os.path.join(self.rootpath, <returned_value>)` for every value from the components

These are just some ideas after having a quick look to the PR, **probably bad ideas** because right now I don't know where the value of the `rootpath` is being used.